File: exemplary_program.cpp
#include <opencv2/imgcodecs.hpp>
#include "superlibrary.h"

int main(int argc, char **argv) {
  // Read demo image
  cv::Mat im = cv::imread("demo_image.png"); 
  
  // Use C++ coolFunction()
  cv::Mat blur = coolFunction(im);
  
  // Use C++ CoolClass::toGray()
  CoolClass cc;
  cv::Mat gray = cc.toGray(im);
  
  // Write output images
  cv::imwrite("blur.png", blur);
  cv::imwrite("gray.png", gray);
}

File: exemplary_program.py
import cv2
import superlibrary

# Read demo image
im = cv2.imread('demo_image.png')

# Use C++ coolFunction()
blur = superlibrary.cool_function(im)

# Use C++ CoolClass::toGray()
cc = superlibrary.CoolClass()
gray = cc.to_gray(im)

# Write output images
cv2.imwrite('blur.png', blur)
cv2.imwrite('gray.png', gray)
